pkg/operation/botanist/monitoring.go

```diff
@@ -740,5 +765,32 @@ func (b *Botanist) DeleteSeedMonitoring(ctx context.Context) error {
 		},
 	}
 
+	// depending on the agentmode setting, prometheus is
+	// either deployed as a StatefulSet or as a Deployment
+	if b.HasPrometheusAgentModeSetting(ctx) && *b.Config.Monitoring.Shoot.AgentMode.Enabled {
+		objects = append(objects, &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "prometheus",
+			},
+		})
+	} else {
+		objects = append(objects, &appsv1.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "prometheus",
+			},
+		})
+	}
+
 	return kutil.DeleteObjects(ctx, b.K8sSeedClient.Client(), objects...)
 }
+
+// If agentMode is enabled, Prometheus is deployed as a Deployment. Otherwise
+// it's a StatefulSet. This function allows easy switching between the two.
+func (b *Botanist) HasPrometheusAgentModeSetting(ctx context.Context) bool {
```

Comment:
This only checks if the config option exists. For simplicity it should also check if AgentMode is enabled?